### ISSUE TITLE
[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] Happens randomly, when trying to pull Product and Product related data

### ISSUE THREAD

**Issue by babinos87** [Shopify/shopify_python_api#665]
Hi, over the last few days we have noticed API errors from Shopify, mainly when we pull product and product related information. 

We are not sure if this is an issue with the package, or with the Shopify servers, or something else.

We are using python 3.10 and the current pyhon package to access the API.

Example errors:

This one occurs when we pull `CustomCollection` entity by id

```
...
File "/home/airflow/.local/.virtualenvs/agents310/lib/python3.10/site-packages/shopify/base.py", line 196, in find
  collection = super(ShopifyResource, cls).find(id_=id_, from_=from_, **kwargs)
File "/home/airflow/.local/.virtualenvs/agents310/lib/python3.10/site-packages/pyactiveresource/activeresource.py", line 386, in find
  return cls._find_every(from_=from_, **kwargs)
File "/home/airflow/.local/.virtualenvs/agents310/lib/python3.10/site-packages/pyactiveresource/activeresource.py", line 525, in _find_every
  response = cls.connection.get(path, cls.headers)
File "/home/airflow/.local/.virtualenvs/agents310/lib/python3.10/site-packages/pyactiveresource/connection.py", line 329, in get
  return self._open('GET', path, headers=headers)
File "/home/airflow/.local/.virtualenvs/agents310/lib/python3.10/site-packages/shopify/base.py", line 23, in _open
  self.response = super(ShopifyConnection, self)._open(*args, **kwargs)
File "/home/airflow/.local/.virtualenvs/agents310/lib/python3.10/site-packages/pyactiveresource/connection.py", line 290, in _open
  raise Error(err, url)
 pyactiveresource.connection.Error: <urlopen error [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:997)>
 ```
 
 We get similar errors when pulling `Product` or `InventoryItem`, for example.
 
 This has started to occur very recently. We have not changed anything to our procedure that pulls data, neither to the software of it.
 
 Some googling indicates that there is some mismatch during the TLS handshake because of unsupported ciphers.
 
 But I am not sure that this is the issue here, because:
 
 * This seems to happen randomly, and mostly when pulling `Product` and `Product`-related data
 * We can pull `Customer` and `Order` data in most cases, if not always, using similar procedure and software
 * I have checked the TLS versions that are supported by our OpenSSL and they seems legitimate (1.3. is usable)
 
 
 Note that the same thing occurs for many different Shopify stores (from different clients of ours), and not just a single store.
 
So, to recap, we cannot pull `Product` and `Product`-related data **SOMETIMES** (the job starts, pulls some data, then we get this error), without a clear indication why this happens.
 
Do you have any ideas / suggestion why this may happen?
 
Could it be something else, irrelevant from the TLS handshake? E.g. some server issue?

Any tips on how to troubleshot this?

Many Thanks!
